CMakeModules/SplitDebugInfo.cmake

```diff
@@ -0,0 +1,88 @@
+# Tailored after https://github.com/GerbilSoft/mcrecover/blob/master/cmake/macros/SplitDebugInformation.cmake
+# Minor modifications to original
+
+if (NOT MSVC)
+  include(CMakeFindBinUtils)
+  if (NOT APPLE AND NOT CMAKE_OBJCOPY)
+    message(WARNING "'objcopy' was not found; debug information will not be split.")
+  elseif (NOT CMAKE_STRIP)
+    message(WARNING "'strip' was not found; debug information will not be split.")
+  endif ()
+endif (NOT MSVC)
+
+function(af_split_debug_info _target)
+  set(SPLIT_TOOL_EXISTS ON)
+  if (MSVC)
+    set(SPLIT_TOOL_EXISTS OFF)
+    if(AF_PACK_DEBUG_SYMBOLS AND WIN32)
+      install(FILES
+        $<TARGET_PDB_FILE:${_target}>
+        DESTINATION ${AF_INSTALL_LIB_DIR}
+        COMPONENT "${_target}_debug_symbols"
+        )
+    endif()
+  elseif (NOT APPLE AND NOT CMAKE_OBJCOPY)
+    set(SPLIT_TOOL_EXISTS OFF)
+  elseif (NOT CMAKE_STRIP)
+    set(SPLIT_TOOL_EXISTS OFF)
+  endif ()
+
+  if (SPLIT_TOOL_EXISTS)
+    get_target_property(TARGET_TYPE ${_target} TYPE)
+    set(PREFIX_EXPR_1
+      "$<$<STREQUAL:$<TARGET_PROPERTY:${_target},PREFIX>,>:${CMAKE_${TARGET_TYPE}_PREFIX}>")
```

Comment:
I think it would be cleaner if this wasn't a generator expression. Generator expressions are usually only useful for IDEs such as visual studio and xcode where you need to change behavior based on the configuration. It doesn't looks like you need that here.

Reply:
I have checked these and this procedure is what works reliably consistently across Windows, OSX and Linux.